### PR TITLE
DEV9: Move ATA write queue logic into its own class

### PR DESCRIPTION
**Pull request by TheLastRar**
This will be useful for porting other CLR_DEV9 features into pcsx2

Also adds the ability for multiple threads to add to the queue, as needed by the DNS pr

### CODE REVIEW

File: pcsx2/DEV9/SimpleQueue.h [hunk 1]
```diff
@@ -0,0 +1,95 @@
+#pragma once
```
Comment (lightningterror) [inline pcsx2/DEV9/SimpleQueue.h:1]: File needs copyright header.

Comment (TheLastRar) [reply pcsx2/DEV9/SimpleQueue.h:16]: Woops, added

File: pcsx2/DEV9/ATA/Commands/ATA_CmdDMA.cpp [hunk 2]
```diff
@@ -62,7 +62,11 @@ void ATA::DRQCmdDMADataFromHost()
 }
 void ATA::PostCmdDMADataFromHost()
 {
-	QueueWrite(currentWriteSectors, currentWrite, currentWriteLength);
+	WriteQueueEntry entry;
+	entry.data = currentWrite;
+	entry.length = currentWriteLength;
+	entry.sector = currentWriteSectors;
+	writeQueue.Enqueue(entry);
```
Comment (refractionpcsx2) [inline pcsx2/DEV9/ATA/Commands/ATA_CmdDMA.cpp:69]: entry.next not initialised

Comment (TheLastRar) [reply pcsx2/DEV9/ATA/Commands/ATA_CmdDMA.cpp:69]: Reiterating what was said in the discord just for reference;

entry is of type WriteQueueEntry (defined within class ATA), this pr removed the next field, so this seems like a false positive.

I noticed that SimpleQueue also defines a structure called WriteQueueEntry (defined within the class) which does have a field called next.

In theory this shouldn't cause issues (as they are defined within there own respective classes), but I've renamed one of them

Comment (TheLastRar) [reply pcsx2/DEV9/ATA/Commands/ATA_CmdDMA.cpp:69]: Zero initializing the whole struct made codacy happy

File: pcsx2/DEV9/SimpleQueue.h [hunk 3]
```diff
@@ -0,0 +1,110 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2020  PCSX2 Dev Team
```
Comment (lightningterror) [inline pcsx2/DEV9/SimpleQueue.h:2]: 2021

Comment (TheLastRar) [reply pcsx2/DEV9/SimpleQueue.h]: Corrected

**Pull request closed without merging**